### PR TITLE
Fix logic for website checker

### PR DESCRIPTION
**Pull request by lkysow**
Workflow should run when no docs/cherry-pick label && no pr/docs-label

### CODE REVIEW

File: .github/workflows/website-checker.yml
Comment (FFMMM) [inline .github/workflows/website-checker.yml:24]:
```suggestion
    # If there's already a `type/docs-cherrypick` and a `pr/no-docs` label, we ignore this check 
```

Comment (lkysow) [reply .github/workflows/website-checker.yml:24]: I think that's not accurate though?

There are three cases of labels:
1. no matching labels => should run workflow
2. type/docs-cherry-pick => shouldn't run workflow
3. pr/no-docs => shouldn't run workflow


The conditional will evaluate to:

1. !false && !false => true && true => true
2. !true && !false => false && true => false
3. !false && !true => true && false => false

So it's accurate to say if there's a docs-cherry-pick *or* a pr-no-docs we ignore the check

Comment (eculver) [reply .github/workflows/website-checker.yml:24]: The intention was to disable the checker in the presence of _either_ label so I think we should keep the comment as is. Otherwise, it would suggest that _both_ labels are required to disable the checker.

**Review by eculver — APPROVED** [review]
The dyslexia definitely played a part in this one. This makes my brain hurt. Thanks for fixing this @lkysow.

File: .github/workflows/website-checker.yml
Comment (lkysow) [reply .github/workflows/website-checker.yml:24]: 👍 I'll merge it but if I'm totally wrong I'm happy to edit that comment.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/614610.

**Review by Protek5150iworld — COMMENTED** [review]
Connect Lequariusjackson0920@protek.pr WIKI R

**Pull request merged**
